### PR TITLE
Improved batch verification for legacy one-of-many proofs

### PR DESCRIPTION
**Pull request by AaronFeickert**
Improves batch verification for legacy one-of-many proofs by reducing to a single multiscalar multiplication operation. Adds additional sanity checks for the verifier routine. Removes redundant code while retaining backwards compatibility.

Review requested.

### CODE REVIEW

**Review by levonpetrosyan93 — APPROVED** [review]

**Comment by levonpetrosyan93** [comment]
LGTM, smoke test done successfully.

File: src/sigma/sigmaplus_verifier.hpp [hunk 1]
```diff
@@ -332,27 +306,6 @@ bool SigmaPlusVerifier<Exponent, GroupElement>::compute_fs(
     return true;
 }
 
-template<class Exponent, class GroupElement>
-bool SigmaPlusVerifier<Exponent, GroupElement>::abcd_checks(
```
Comment (levonpetrosyan93) [inline src/sigma/sigmaplus_verifier.hpp:336]: Please also remove function declaration from header.

**Comment by levonpetrosyan93** [comment]
Here are the performance numbers on my machine
 N = 16384 n = 4 m = 7 M = 1
 Verify time New 232 ms 
 Verify time Old 233 ms 
 N = 16384 n = 4 m = 7 M = 5
 Verify time New 245 ms 
 Verify time Old 251 ms 
 N = 16384 n = 4 m = 7 M = 10
 Verify time New 273 ms 
 Verify time Old 287 ms 
 N = 16384 n = 4 m = 7 M = 50
 Verify time New 397 ms 
 Verify time Old 473 ms 
 N = 16384 n = 4 m = 7 M = 100
 Verify time New 534 ms 
 Verify time Old 621 ms 
 N = 16384 n = 4 m = 7 M = 500
 Verify time New 1812 ms 
 Verify time Old 2173 ms 
 N = 16384 n = 4 m = 7 M = 1000
 Verify time New 3535 ms 
 Verify time Old 4267 ms

File: src/sigma/sigmaplus_verifier.hpp (hunk 1 above)
Comment (AaronFeickert) [reply src/sigma/sigmaplus_verifier.hpp:336]: Done.

**Review by levonpetrosyan93 — DISMISSED** [review]

**Review by levonpetrosyan93 — APPROVED** [review]

**Pull request merged**